### ISSUE TITLE
There should be an explicit option to exclude documents or schemas

### ISSUE THREAD

**Issue by slikts** [dotansimha/graphql-code-generator#1698]
Related to #1229

Currently the way to exclude files is by using glob patterns (for example, like so: `src/**/!(*.spec).{tsx,graphql}`), but it's complex and doesn't scale well with the number of exceptions that may be necessary in more legacy code bases. It would be useful if there was a separate config option to exclude files; something like:
```yaml
documents:
  - "src/**/*.{tsx,graphql}"
  exclude: "*.spec.*"
```

**Comment by dotansimha** [comment]
I think we can switch to use this package: https://www.npmjs.com/package/globby , and then we'll be able to provide multiple globs, and it will be easy to do something like:
```
documents:
  - "src/**/*.{tsx,graphql}"
  - "!*.spec.*"
```

What do you think @slikts ?

**Comment by slikts** [comment]
Sure, I don't see a problem with that; the issue with the current approach is the complexity of having multiple or split sub-patterns, which would be solved with multiple patterns. A secondary issue is that inverse glob patterns aren't that well known, so `docs/getting-started/documents-field.md` should feature an example.

**Comment by dotansimha** [comment]
Implemented in: https://github.com/dotansimha/graphql-code-generator/pull/2283

**Comment by dotansimha** [comment]
Available in 1.5.0 🎉
